test/uk/gov/hmrc/submitpublicpensionadjustment/viewmodels/PDFViewModelSpec.scala

```diff
@@ -68,52 +68,128 @@ class PDFViewModelSpec extends AnyFreeSpec with Matchers with Logging {
         dob = "13/01/1920",
         addressLine1 = "testLine1",
         addressLine2 = "testLine2",
-        postCode = "Postcode",
-        country = None,
-        utr = None,
+        townOrCity = "TestCity",
```

Comment:
This test exercises all of the view model building code in one go, and whilst it may pass it is difficult to see which mapping scenarios it is exercising. Does it exercise all the code paths?

A top level test may be useful, but section based tests would be more readable and easier to adapt in the event that the some of the sections need to change.